backend/pkg/api/handle_kafka_connect_integration_test.go

```diff
@@ -0,0 +1,402 @@
+				"value.converter":                           "org.apache.kafka.connect.json.JsonConverter",
+				"value.converter.schemas.enable":            "false",
+			},
+		}
+
+		res, body := s.apiRequest(ctx, http.MethodPost, "/api/kafka-connect/clusters/redpanda_connect/connectors", input)
+
+		require.Equal(200, res.StatusCode)
+
+		createConnectRes := con.ConnectorInfo{}
+		err := json.Unmarshal(body, &createConnectRes)
+		assert.NoError(err)
+
+		assert.Equal("http_connect_input", createConnectRes.Name)
+		assert.Equal("httpbin-input", createConnectRes.Config["kafka.topic"])
+		assert.Equal("1000", createConnectRes.Config["http.timer.interval.millis"])
+
+		// timeout to allow for connect instance to perform few requests
+		timer := time.NewTimer(3500 * time.Millisecond)
+		<-timer.C
+
+		cl, err := kgo.NewClient(
+			kgo.SeedBrokers(seedBroker),
+			kgo.ConsumeTopics("httpbin-input"),
+			kgo.ConsumeResetOffset(kgo.NewOffset().AtStart()),
+		)
+		require.NoError(err)
+
+		defer cl.Close()
+
+		type uuidValue struct {
+			UUID string `json:"uuid"`
+		}
+
+		type recordValue struct {
+			Value string `json:"value"`
+			Key   string
+		}
+
+		records := make([][]byte, 0)
+
+		// control polling end via context
+		pollCtx, pollCancel := context.WithTimeout(context.Background(), 3*time.Second)
+		defer pollCancel()
+
+		// check the data in the sink topic
+		for {
+			fetches := cl.PollFetches(pollCtx)
+			if fetches.IsClientClosed() {
+				break
+			}
+
+			errs := fetches.Errors()
+			if len(errs) == 1 && (errors.Is(errs[0].Err, context.DeadlineExceeded) || errors.Is(errs[0].Err, context.Canceled)) {
+				break
+			}
+
+			require.Empty(errs)
+
+			fetches.EachRecord(func(record *kgo.Record) {
+				records = append(records, record.Value)
+			})
+		}
+
+		for _, vd := range records {
+			rv := recordValue{}
+			err := json.Unmarshal(vd, &rv)
+			assert.NoError(err)
+			assert.NotEmpty(rv.Value)
+
+			uv := uuidValue{}
+			err = json.Unmarshal([]byte(rv.Value), &uv)
+			assert.NoError(err)
+
+			_, err = uuid.Parse(uv.UUID)
+			assert.NoError(err)
+		}
+
+		nRecords := len(records)
+		assert.True(nRecords >= 3 && nRecords <= 5, "expect to have between 3 and 5 records. got: %d", nRecords)
+
+		// check GET
+		getRes, getBody := s.apiRequest(context.Background(), http.MethodGet, "/api/kafka-connect/clusters/redpanda_connect/connectors/http_connect_input", nil)
+		require.Equal(200, getRes.StatusCode)
+
+		getConnectRes := connect.ClusterConnectorInfo{}
+		err = json.Unmarshal(getBody, &getConnectRes)
+		assert.NoError(err)
+
+		assert.Equal("http_connect_input", getConnectRes.Name)
+		assert.Equal("httpbin-input", getConnectRes.Config["kafka.topic"])
+		assert.Equal("1000", getConnectRes.Config["http.timer.interval.millis"])
+	})
+}
+
+const testConnectConfig = `key.converter=org.apache.kafka.connect.converters.ByteArrayConverter
+value.converter=org.apache.kafka.connect.converters.ByteArrayConverter
```

Comment:
if you specify key/value.converter in the connector config, it I think can be removed from here as well

Reply:
When I try to remove these I get the following error on startup:
```
{"ts":"2023-07-17 12:18:28,043","level":"ERROR","thread":"main","class":"org.apache.kafka.connect.cli.ConnectDistributed:89","msg":"Stopping due to error"}
org.apache.kafka.common.config.ConfigException: Missing required configuration "key.converter" which has no default value.
	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:493)
	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:483)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:112)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:132)
	at org.apache.kafka.connect.runtime.WorkerConfig.<init>(WorkerConfig.java:475)
	at org.apache.kafka.connect.runtime.distributed.DistributedConfig.<init>(DistributedConfig.java:560)
	at org.apache.kafka.connect.runtime.distributed.DistributedConfig.<init>(DistributedConfig.java:555)
	at org.apache.kafka.connect.cli.ConnectDistributed.startConnect(ConnectDistributed.java:98)
	at org.apache.kafka.connect.cli.ConnectDistributed.main(ConnectDistributed.java:83)
```

So leaving them for now.